### PR TITLE
SONAR-8838 Compute tag distribution for WS api/components/search_projects

### PR DESCRIPTION
**Pull request by teryk**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/measure/index/ProjectMeasuresIndex.java [hunk 1]
```diff
@@ -151,7 +153,7 @@ private static void addMetricSort(ProjectMeasuresQuery query, SearchRequestBuild
         .order(query.isAsc() ? ASC : DESC));
   }
 
-  private static void addFacets(SearchRequestBuilder esSearch, SearchOptions options, Map<String, QueryBuilder> filters) {
+  private static void addFacets(ProjectMeasuresQuery query, SearchRequestBuilder esSearch, SearchOptions options, Map<String, QueryBuilder> filters) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/measure/index/ProjectMeasuresIndex.java]: MAJOR Remove this unused method parameter "query". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1172)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: server/sonar-server/src/test/java/org/sonar/server/measure/index/ProjectMeasuresIndexTest.java [hunk 2]
```diff
@@ -1029,6 +1030,52 @@ public void facet_languages_contains_only_projects_authorized_for_user() throws
...
+  @Test
+  public void facet_tags_is_sticky() {
+    index(
+      newDoc().setTags(newArrayList("finance")),
+      newDoc().setTags(newArrayList("finance")),
+      newDoc().setTags(newArrayList("cpp")));
+
+    Map<String, Long> result = underTest.search(new ProjectMeasuresQuery().setTags(newHashSet("cpp")), new SearchOptions().addFacets(FIELD_TAGS)).getFacets().get(FIELD_TAGS);
+
+    assertThat(result).containsOnly(
```
Comment (teryk) [inline server/sonar-server/src/test/java/org/sonar/server/measure/index/ProjectMeasuresIndexTest.java]: I filter on tags that must be equals to cpp. I check that finance is equal to 2. What am I missing?

Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/measure/index/ProjectMeasuresIndexTest.java]: Oh sorry, I didn't see the filter on tags. 
In fact to check stickyness I like to add another data in the doc and check that : 

1. the filter is not taking into account on the tags facet
2. the filter is taking into account on the other facet 

It makes test clearer to me, but it's a personal opinion. 
It's true that your test is testing the stickyness

**Pull request merged**